### PR TITLE
Whens are executed after the endpoint is started, removed Given in fa…

### PR DESCRIPTION
**Pull request by danielmarbach**
As discussed today with @andreasohlund @SzymonPobiega and @timbussmann the following behavior is now changed

* Endpoints are still started asynchronously on the ThreadPool
* The endpoints are started and then
* The `EndpointsStarted` flag is set on the `ScenarioContext` and only after that
* Whens are executed
* Also removed `Givens`

Please review

### CODE REVIEW

File: src/NServiceBus.AcceptanceTests/Audit/When_auditing_message_with_TimeToBeReceived.cs [hunk 1]
```diff
@@ -13,7 +13,7 @@ public class When_auditing_message_with_TimeToBeReceived : NServiceBusAcceptance
         public async Task Should_not_honor_TimeToBeReceived_for_audit_message()
         {
             var context = await Scenario.Define<Context>()
-            .WithEndpoint<EndpointWithAuditOn>(b => b.Given(bus => bus.SendLocalAsync(new MessageToBeAudited())))
+            .WithEndpoint<EndpointWithAuditOn>(b => b.When(bus => bus.SendLocalAsync(new MessageToBeAudited())))
```
Comment (andreasohlund) [inline src/NServiceBus.AcceptanceTests/Audit/When_auditing_message_with_TimeToBeReceived.cs]: `When` doesn't read so good without a condition. Should we consider splitting into

`WhenStarted(bus=>)` and `When(condition, bus=>)`  ?

Comment (timbussmann) [inline src/NServiceBus.AcceptanceTests/Audit/When_auditing_message_with_TimeToBeReceived.cs]: I think it reads better. Given -> When -> Then is quite common for spec testing frameworks. 
`Given` in our case would correspond to `WithEndpoint`, so the tests would read "Given *endpoint x* When *sending a message* Then *assertions* which is much cleaner.

Comment (danielmarbach) [inline src/NServiceBus.AcceptanceTests/Audit/When_auditing_message_with_TimeToBeReceived.cs]: @andreasohlund we discussed that before and agreed on not doing it

Comment (SzymonPobiega) [inline src/NServiceBus.AcceptanceTests/Audit/When_auditing_message_with_TimeToBeReceived.cs]: No strong opinion here

Comment (andreasohlund) [inline src/NServiceBus.AcceptanceTests/Audit/When_auditing_message_with_TimeToBeReceived.cs]: Ok I see it now, When essentially means "When all endpoints are started successfully". I'm good with the current design

**Pull request merged**